stratum/lib/security/certificate.cc

```diff
@@ -54,43 +63,42 @@ util::Status GenerateRSAKeyPair(EVP_PKEY* evp, int bits) {
   BIGNUM_ptr exp(BN_new(), BN_free);
   BN_set_word(exp.get(), RSA_F4);
   RSA* rsa = RSA_new();
-  CHECK_RETURN_IF_FALSE(RSA_generate_key_ex(rsa, bits, exp.get(), NULL))
+  CHECK_RETURN_IF_FALSE(RSA_generate_key_ex(rsa, bits, exp.get(), nullptr))
       << "Failed to generate RSA key.";
-
   // Store this keypair in evp
   // It will be freed when EVP is freed, so only free on failure.
   if (!EVP_PKEY_assign_RSA(evp, rsa)) {
     RSA_free(rsa);
     RETURN_ERROR(ERR_INVALID_PARAM) << "Failed to assign key.";
   }
+
   return util::OkStatus();
 }
 
 util::Status GenerateUnsignedCert(X509* unsigned_cert,
                                   EVP_PKEY* unsigned_cert_key,
                                   const std::string& common_name, int serial,
-                                  int days) {
+                                  absl::Time valid_until) {
```

Comment:
```suggestion
                                  absl::Time issued_time,
                                  absl::Duration valid_duration) {
```

Not sure if we should have valid_until or a duration if we also include a start time param.

Reply:
Creating durations longer than hours is a bit involved, as the `absl::<time unit>` helpers stop at `Hours`, for good reason. Absolute `Time` points (`valid_after`, `valid_util` ?) seem intuitive to me.

Reply:
Absolute times work for me.